pkg/controllers/provisioningv2/rke2/bootstrap/controller.go

```diff
@@ -131,7 +136,25 @@ func (h *handler) getBootstrapSecret(namespace, name string, envVars []corev1.En
 	if os := machine.GetLabels()[rke2.CattleOSLabel]; os == rke2.WindowsMachineOS {
 		is = installer.WindowsInstallScript
 	}
-	data, err := is(context.WithValue(context.Background(), tls.InternalAPI, hasHostPort), base64.URLEncoding.EncodeToString(hash[:]), envVars, "")
+
+	// find the machine pool which the current machine belongs to
+	// so that we can utilize any provided hooks.
+	provCluster, err := h.provClusters.Get(namespace, machine.Spec.ClusterName)
```

Comment:
I feel like directly accessing the provisioning cluster may not be the most appropriate way to do this. It's the simplest, but if we want to stick closer to upstream, we would refactor this to create a bootstrap template per pool: https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/provisioningv2/rke2/provisioningcluster/template.go#L235. cc @Oats87

Reply:
Yes, the most ideal manner of fixing this is to create a bootstrap template per pool; however, if we make this change we need to figure out how to make the change in such a way where we do not cause reprovisioning of existing machine pools when we change the name of the bootstrap template, and we need to clean up the old bootstrap templates. If possible, I really would like to clean up that behavior.

I'm not sure if making that change makes the scope of adding these prehooks too big for the given milestone though, but I would much prefer this method. 

cc: @snasovich